src/client/utils.ts

```diff
@@ -1,18 +1,29 @@
-import { Range, TextDocument, window, workspace } from 'vscode';
+import {
+	Range,
+	TextDocument,
+	workspace,
+	Uri,
+	WorkspaceConfiguration
+} from "vscode";
 
-const defaultCommand = 'v';
+const defaultCommand = "v";
 
 export function fullDocumentRange(document: TextDocument): Range {
 	const lastLineId = document.lineCount - 1;
 	return new Range(0, 0, lastLineId, document.lineAt(lastLineId).text.length);
 }
 
 export function getVExecCommand(args: string): string {
-	const config = workspace.getConfiguration('v');
-	const vPath = config.get('pathToExecutableFile', '') || defaultCommand;
+	const config = getVConfig();
```

Comment:
Should `getVConfig` take an argument here?

Reply:
Maybe... but optional...

Reply:
When getting config, we must pass the URI of the document, so VSCode can get the nearest config on the workspace because on VSCode, the user can have workspace config right. If we don't pass a URI of the document, VSCode will get the global config.

Reply:
Maybe do this later.

Reply:
The folder config that I mean is `.vscode/settings.json`

Reply:
I have done an experiment and the results are as I expected.

- Change global config:
```json
...
"v.format.args": "-workspace"
...
``` 


- Make a folder config. It's on `.vscode/settings.json`:
```json
...
"v.format.args": "-folder"
...
```

---------------------------------------------------------------------------------
**Test 1**
- Change `format.ts` (With no URI specified) :
```ts
...
const vfmtArgs = getVConfig().get("format.args", "");
console.log(vfmtArgs)
...
```
- Open `main.v`.
- Format the `main.v`.
- The result is `-workspace`


---------------------------------------------------------------------------------
**Test 2**
- Change `format.ts` (With URI specified) :
```ts
...
const vfmtArgs = getVConfig(document.uri).get("format.args", "");
console.log(vfmtArgs)
...
```
- Open `main.v`.
- Format the `main.v`.
- The result is `-folder`

Reply:
So yes... the `getVConfig`should have URI argument but for now, just optional.